src/pages/Shop/index.js

```diff
@@ -0,0 +1,372 @@
+import React, { Component } from "react";
+import styles from "./shop.module.css";
+import { connect } from "react-redux";
+import { getListProducts } from "../../action/shop";
+import { addToCart } from "../../action/cart";
+import Loading from "../../components/Loading";
+import {
+  MIN_VIEW_SHOP,
+  AVERAGE_VIEW_SHOP,
+  MAX_VIEW_SHOP
+} from "../../constants";
+import TriangleTop from "../../icons/TriangleTop";
+import Basket from "../../icons/Basket";
+import { isEmpty } from "../../utils/isEmpty";
+import { Link } from "react-router-dom";
+import { stringifySearchParams } from "../../utils/stringifySearchParams";
+
+class Shop extends Component {
+  state = {
+    dropViewIsOpen: false,
+    dropDateIsOpen: false,
+    dropPriceIsOpen: false
```

Comment:
maybe you can use only one state for all your dropDown view??
 for example
```
    dropDownModalStatus: null 
```

setState({dropDownModalStatus: 'VIEW'})
setState({dropDownModalStatus: 'PRICE'})
setState({dropDownModalStatus: "CREATED_AT"})